src/main.rs

```diff
@@ -30,14 +38,30 @@ fn main() {
     let html_dir = matches.value_of(ARG_OUTPUT_DIRECTORY).unwrap_or(DEFAULT_HTML_DIR);
 
     // Init logger crate
-    mowl::init().unwrap();
+    match mowl::init() {
+        Ok(_) => debug!("Mowl logging initiated."),
+        Err(_) => {
+            retval.code = ErrorType::InitFailure;
+            return retval;
+        },
+    }
 
     // This can be deleted when html_dir is used further
     debug!("Output path: {}", html_dir);
 
     // Do first processing steps
     let mut processing = Processing::default();
-    processing.read_from_directory(&md_dir);
+
+    return_if_not_ok!(processing.read_from_directory(&md_dir));
```

Comment:
A I just saw that there is already a run function. The return type should be the above Result after the implementation of the `Error` trait for the structure.